components/email-mgt/org.wso2.carbon.email.mgt/src/main/java/org/wso2/carbon/email/mgt/NotificationTemplateManagerImpl.java

```diff
@@ -0,0 +1,720 @@
+                    TemplateMgtConstants.ErrorMessages.ERROR_CODE_ERROR_RESOLVING_MAIN_APPLICATION.getCode(),
+                    TemplateMgtConstants.ErrorScenarios.NOTIFICATION_TEMPLATE_MANAGER);
+            String message = String.format(
+                    TemplateMgtConstants.ErrorMessages.ERROR_CODE_ERROR_RESOLVING_MAIN_APPLICATION.getMessage(),
+                    applicationUuid, tenantDomain);
+            throw new NotificationTemplateManagerException(code, message, e);
+        }
+        validateTemplateLocale(locale);
+        locale = normalizeLocaleFormat(locale);
+        validateDisplayNameOfTemplateType(templateType);
+        assertTemplateTypeExists(templateType, notificationChannel, tenantDomain);
+        NotificationTemplate notificationTemplate = userDefinedTemplatePersistenceManager.getNotificationTemplate(
+                templateType, locale, notificationChannel, applicationUuid, tenantDomain);
+
+        String defaultLocale = getDefaultNotificationLocale(notificationChannel);
+        if (notificationTemplate == null) {
+            if (StringUtils.equalsIgnoreCase(locale, defaultLocale)) {
+                // Template is not available in the default locale. Therefore, breaking the flow at the consuming side
+                // to avoid NPE.
+                String code = I18nEmailUtil.prependOperationScenarioToErrorCode(
+                        TemplateMgtConstants.ErrorMessages.ERROR_CODE_TEMPLATE_NOT_FOUND.getCode(),
+                        TemplateMgtConstants.ErrorScenarios.NOTIFICATION_TEMPLATE_MANAGER);
+                String errorMessage = String
+                        .format(TemplateMgtConstants.ErrorMessages.ERROR_CODE_TEMPLATE_NOT_FOUND.getMessage(),
+                                templateType, tenantDomain);
+                throw new NotificationTemplateManagerServerException(code, errorMessage);
+            } else {
+                if (log.isDebugEnabled()) {
+                    String message = String
+                            .format("'%s' template in '%s' locale was not found in '%s' tenant. Trying to return the "
+                                            + "template in default locale : '%s'", templateType, locale, tenantDomain,
+                                    defaultLocale);
+                    log.debug(message);
+                }
+                // Try to get the template type in default locale.
+                return getNotificationTemplate(notificationChannel, templateType, defaultLocale, tenantDomain);
+            }
+        }
+        return notificationTemplate;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<NotificationTemplate> getAllSystemNotificationTemplatesOfType(String notificationChannel,
+                                              String templateDisplayName) throws NotificationTemplateManagerException {
+
+        assertSystemTemplateTypeExists(templateDisplayName, notificationChannel);
```

Comment:
Rename to a better function name.

Reply:
addressed with https://github.com/wso2-extensions/identity-event-handler-notification/pull/250/commits/4a4dae818d530081e8a91a33d6d70da13c375c9c